app/services/HttpIdentityService.scala

```diff
@@ -78,6 +80,17 @@ class HttpIdentityService(apiUrl: String, apiClientToken: String)(implicit wsCli
     }
   }
 
+  def setPasswordGuest(
+    password: String,
+    guestAccountRegistrationToken: String
+  )(implicit ec: ExecutionContext): EitherT[Future, IdentityApiResponseError, CookiesResponse] = {
+    val payload = Json.obj("password" -> password)
+    val headers =
+      List("Authorization" -> s"Bearer $apiClientToken", "X-Guest-Registration-Token" -> guestAccountRegistrationToken, "Content-Type" -> "application/json")
+    val urlParameters = List("validate-email" -> "0")
+    EitherT(request(s"guest/password", headers, urlParameters).put(payload) map JsonBodyParser.extract[CookiesResponse](JsonBodyParser.jsonField("cookies")))
```

Comment:
It's preferable to call `attemptT` on the returned future since this will take any throwables on the executing thread and wrap them in a `Left()` i.e. the `Future` is pretty much guaranteed not to have a `Failure`; whereas with this constructor, you could have a `Left()` or `Right()` or a`Failure` in the `Future`.

Reply:
Minor, but I think methods without parentheses and dots was a bad idea (Odersky himself has now said he's not a fan). Think `.map()` is preferable, although some people may disagree...